web/src/lib/components/StoreView.svelte

```diff
@@ -0,0 +1,117 @@
+<script lang="ts">
+    import PublicAuctionCard from "../components/PublicAuctionCard.svelte";
+    import type { Auction } from "../types/auction";
+    import type { User } from "../types/user";
+    import { type ILoader, getEntities, getStore, getProfile, ErrorHandler } from "../services/api";
+    import { onMount } from "svelte";
+    import { token, user } from "../stores";
+    import StoreNotFound from "./StoreNotFound.svelte";
+    import { afterNavigate } from '$app/navigation';
+    import type { IEntity } from '$lib/types/base';
+
+    export let storeName;
+    export let loader: ILoader;
+    export let entities: IEntity[] | null = null;
+
+    let auctions: Auction[] | null = null;
+    let store: User;
+    let currentTab = "ACTIVE AUCTIONS";
+    let loading = true;
+
+    function fetchProfile(tokenValue) {
+        getProfile(tokenValue,
+            u => {
+                user.set(u);
```

Comment:
The only component "loading" the profile is the Navbar, everybody else should use the user object already cached in the store and/or .subscribe() to the user.

We don't want random components to keep updating our profile, which acts like a "global" - plus, the profile doesn't change - it changes on login and when edited. That's it. In rest it should be read only.